packages/components/src/internal/components/samples/SampleDeleteMenuItem.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, memo, useCallback, useState } from 'react';
+import { MenuItem } from 'react-bootstrap';
+
+import { AuditBehaviorTypes } from '@labkey/api';
+
+import { EntityDeleteModal, SampleTypeDataType, QueryModel, SelectionMenuItem } from '../../..';
+
+interface Props {
+    queryModel: QueryModel;
+    key?: string;
+    itemText?: string;
+    verb?: string;
+    beforeSampleDelete?: () => any;
+    afterSampleDelete?: (rowsToKeep?: any[]) => any;
+    auditBehavior?: AuditBehaviorTypes;
+    maxDeleteRows?: number;
+}
+
+export const SampleDeleteMenuItem: FC<Props> = memo(props => {
+    const {
+        key,
+        itemText,
+        queryModel,
+        verb,
+        beforeSampleDelete,
+        afterSampleDelete,
+        auditBehavior,
+        maxDeleteRows,
+    } = props;
+    const [showConfirmDeleteSamples, setShowConfirmDeleteSamples] = useState<boolean>(false);
+
+    const onClick = useCallback(() => {
+        if (!queryModel || queryModel.selections.size > 0) {
```

Comment:
I believe instead of the `queryModel.selections.size > 0` check, there is a helper for `queryModel.hasSelections`

Reply:
What is the scenario were we want to allow for clicking on the menu item to show the modal when `!queryModel`?

Reply:
Ah, I see. The use case is likely for the usage of this menu item component for a single sample details page. Nevermind then.